packages/analytics/add-cloud-link-handler/index.ts

```diff
@@ -0,0 +1,39 @@
+import { getProductIntentFromURL } from '../get-product-intent-from-url'
+import { getUTMParamsCaptureState } from '../utm-params-capture'
+
+const destinations: string[] = [
+  'portal.cloud.hashicorp.com',
+  'app.terraform.io',
+]
+const containsDestination = (str: string): boolean =>
+  destinations.some(function (destination) {
+    return str.indexOf(destination) >= 0
+  })
+
+export function addCloudLinkHandler() {
+  if (typeof window === 'undefined') return
+
+  window.addEventListener('click', (event) => {
+    const linkElement = (event.target as HTMLElement).closest('a')
+    if (linkElement && containsDestination(linkElement.href)) {
+      event.preventDefault()
+      const productIntent = getProductIntentFromURL()
+      const searchParams = {
+        ...getUTMParamsCaptureState(),
+        ...(productIntent && {
+          product: productIntent,
+        }),
+      }
+      try {
+        const url = new URL(linkElement.href)
+        location.href = `${url.origin}${
+          Object.keys(searchParams).length > 0
+            ? `?${new URLSearchParams(searchParams).toString()}`
+            : ''
+        }`
+      } catch (error) {
+        console.error(error)
```

Comment:
Wonder if we should fallback to the `linkElement.href` here in the catch block?

```suggestion
        window.location = linkElement.href
        console.error(error)
```

Reply:
I think that would be a good idea! Better to degrade gracefully especially in a situation like this where we hijack the default behavior.

Reply:
https://github.com/hashicorp/web-platform-packages/pull/72/commits/13237d9f41795083a2e47b2a1b1cba6e45458bbd